gradio/components/image.py

```diff
@@ -249,7 +249,7 @@ def postprocess(
         elif isinstance(y, _Image.Image):
             path = processing_utils.save_pil_to_cache(y, cache_dir=self.GRADIO_CACHE)
         elif isinstance(y, (str, Path)):
-            path = y if isinstance(y, str) else y.name
+            path = y if isinstance(y, str) else str(y)
```

Comment:
We should have always returned the full path. Bug on my part.

Reply:
```suggestion
            path = str(y)
```